awsiot/greengrass_discovery.py

```diff
@@ -44,6 +44,11 @@ def __init__(
         self._socket_options = socket_options
         self._region = region
         self._gg_server_name = 'greengrass-ats.iot.{}.amazonaws.com'.format(region)
+        
+        #: Temporary fix for connection with china endpoint
+        if region == 'cn-north-1':
+            self._gg_server_name = 'greengrass.ats.iot.{}.amazonaws.com.cn'.format(region)
```

Comment:
Perhaps a better solution is to let the user pass the whole endpoint in, like we do with the IoT Core endpoint.

Reply:
Can we add an optional endpoint parameter _in addition to_ making this work out of the box?

Reply:
Do you think it should work for all endpoints out of the box? Because it will still work for almost all endpoints out of the box, just for special ones(like china) the user just needs to pass in the endpoint they need